### PR TITLE
feat: toggle visible for bot sources

### PR DESCRIPTION
**Pull request by datphamcode295**
### Description
- [x] toggle visible for bot sources

### CODE REVIEW

File: src/server/api/routers/bot-source.ts [hunk 1]
```diff
@@ -51,6 +52,33 @@ function deleteByIdHandler() {
     })
 }
 
+function toggleVisibility() {
```
Comment (taynguyen) [inline src/server/api/routers/bot-source.ts:55]: Let's not define it toggle. 
Let's define function setVisibility
FE would set it true/false directly instead.

File: src/server/api/routers/bot-source.ts [hunk 2]
```diff
@@ -51,6 +52,33 @@ function deleteByIdHandler() {
...
+      }),
+    )
+    .mutation(async ({ input }) => {
+      const bs = await db
+        .select({
+          id: schema.botSources.id,
+          visible: schema.botSources.visible,
+        })
+        .from(schema.botSources)
+        .where(eq(schema.botSources.id, input.botSourceId))
```
Comment (taynguyen) [inline src/server/api/routers/bot-source.ts:69]: Let's also select where created_by == ctx.session.user.id
to ensure bot's authentication.

File: src/server/api/routers/bot-source.ts (hunk 1 above)
Comment (datphamcode295) [reply src/server/api/routers/bot-source.ts]: updated

File: src/server/api/routers/bot-source.ts [hunk 3]
```diff
@@ -51,6 +52,38 @@ function deleteByIdHandler() {
...
+            eq(schema.botSources.createdBy, ctx.session.user.id),
+          ),
+        )
+      if (!bs.length) {
+        throw new Error('Bot source not found')
+      }
+
+      await db
+        .update(schema.botSources)
+        .set({ visible: input.visible })
```
Comment (taynguyen) [inline src/server/api/routers/bot-source.ts:82]: Let's also update updated_at and updated_by

File: src/server/api/routers/chat.ts [hunk 4]
```diff
@@ -189,7 +189,12 @@ async function buildPrompt(
       schema.botSources,
       eq(schema.botSources.id, schema.botSourceExtractedData.botSourceId),
     )
-    .where(eq(schema.botSources.botId, bot.id))
+    .where(
+      and(
+        (eq(schema.botSources.botId, bot.id),
```
Comment (taynguyen) [inline src/server/api/routers/chat.ts:194]: Seem don't need parenthesis pair here.

**Review by taynguyen — APPROVED** [review]
LGTM

**Pull request merged**